src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -143,6 +159,13 @@ Status CoreWorkerMemoryStore::Put(const ObjectID &object_id, const RayObject &ob
       object_async_get_requests_.erase(async_callback_it);
     }
 
+    auto promoted_it = promoted_to_plasma_.find(object_id);
+    if (promoted_it != promoted_to_plasma_.end()) {
+      RAY_CHECK(store_in_plasma_ != nullptr);
+      store_in_plasma_(object, object_id.WithTransportType(TaskTransportType::RAYLET));
```

Comment:
Shouldn't we update the object that gets added to the in-memory store here so that we're only storing an error to look in plasma? Maybe I'm just confused; is that already being done somewhere?

Reply:
We could do the update, but it isn't needed exactly (it's not harmful to have the value in two places, and could be good for performance in some situations).